File: functions/alsa-lib/variants/go/2_muldiv_down_8.go
package main

func muldivDown(a, b, c int32) int32 {
	v := int64(a) * int64(b) / int64(c)
	if v > 2147483647 {
		return 2147483647
	} else if v < -2147483648 {
		return -2147483648
	}
	return int32(v)
}


func main(){
    muldivDown(0,0,0)
}

File: functions/alsa-lib/variants/c/2_muldiv_down_8.c
#include <stdint.h>

static inline int muldiv_down(int a, int b, int c)
{
	int64_t v = ((int64_t)a * b) / (int64_t)c;
	int maxInt = 2147483647;
	int minInt = -2147483648;
	if (v > maxInt) {
		return maxInt;
	}
	if (v < minInt) {
		return minInt;
	}
	return (int)v;
}


int main(){
    muldiv_down(0,0,0);
}